### PR TITLE
Models that require Contact or Schedule to be built on them, take care of this themselves

### PR DESCRIPTION
**Pull request by casaper**
Just soing something that has always hurt me, when I looked at the thing. SCNR.

### CODE REVIEW

File: app/controllers/clients_controller.rb [hunk 1]
```diff
@@ -58,6 +56,6 @@ def client_params
       :state, :comments, :involved_authority, :competent_authority, :actual_activities,
       language_skills_attributes, relatives_attributes, schedules_attributes,
       contact_attributes
-      )
+    )
```
Comment (casaper) [inline app/controllers/clients_controller.rb]: I know, its unrelated. But rubocop hassles us every time and again. And it should have never been committed unlinted anyway...

File: app/controllers/profiles_controller.rb [hunk 2]
```diff
@@ -44,6 +43,6 @@ def profile_params
       :user_id, :picture, :profession, :monday,
       :tuesday, :wednesday, :thursday, :friday, :avatar,
       contact_attributes
-      )
+    )
```
Comment (casaper) [inline app/controllers/profiles_controller.rb]: I know, it's unrelated. But rubocop hassles us every time and again. And it should have never been committed un-linted anyway...

File: app/models/concerns/build_contact_relation.rb [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+  included do
+    after_initialize :build_contact_relation
+
+    # Ensures that the model has its required relation.
+    #
+    # It makes doing this in every controller new action obsolete.
+    # The model takes care of its requirements itsself, as it should.
+    def build_contact_relation
+      self.contact = Contact.new unless contact
```
Comment (toupeira) [inline app/models/concerns/build_contact_relation.rb]: Simplified: `build_contact unless contact`

File: app/models/concerns/build_schedules_relation.rb [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+
+  included do
+    after_initialize :build_schedules_relation
+
+    # Ensures that the required Schedules are present in the model.
+    #
+    # It makes the need to do this in controller new actions obsolete.
+    # The model takes care of its requirements itsself, as it should.
+    def build_schedules_relation
+      schedules << Schedule.build unless schedules.any?
```
Comment (toupeira) [inline app/models/concerns/build_schedules_relation.rb]: Simplified: `schedules.build unless schedules.any?`

Comment (casaper) [inline app/models/concerns/build_schedules_relation.rb]: Will this `schedules.build` really do the same as `schedules << Schedule.build`?

File: app/models/concerns/build_contact_relation.rb (hunk 3 above)
Comment (casaper) [inline app/models/concerns/build_contact_relation.rb]: But here you're right.

File: app/models/concerns/build_schedules_relation.rb (hunk 4 above)
Comment (toupeira) [inline app/models/concerns/build_schedules_relation.rb]: Ah sorry I thought `#build` was a standard AR method, in that case ignore my comment :)

File: test/models/client_test.rb [hunk 5]
```diff
@@ -26,4 +26,14 @@ def setup
     result = Client.need_accompanying
     assert_equal [], result.to_a
   end
+
+  test 'schedules build correctly automaticly' do
```
Comment (toupeira) [inline test/models/client_test.rb]: `automatically` 👮 

(I'm a stickler for correct grammar, feel free to ignore any comments from me with a 👮)

File: test/models/client_test.rb [hunk 6]
```diff
@@ -26,4 +26,14 @@ def setup
...
+    new_client = Client.new
+    assert_equal new_client.schedules.size, 21
+  end
+
+  test 'contact relation is build automaticly' do
```
Comment (toupeira) [inline test/models/client_test.rb]: `built automatically` 👮

Comment (toupeira) [inline test/models/client_test.rb]: Also in the other specs :)

**Pull request merged**